hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacityScheduler.java

```diff
@@ -3366,14 +3367,47 @@ public boolean moveReservedContainer(RMContainer toBeMovedContainer,
 
   @Override
   public long checkAndGetApplicationLifetime(String queueName,
-      long lifetimeRequestedByApp) {
-    readLock.lock();
+                                             long lifetimeRequestedByApp, RMAppImpl app) {
+    CSQueue queue;
+
+    writeLock.lock();
     try {
-      CSQueue queue = getQueue(queueName);
+      queue = getQueue(queueName);
+
+      // This handles the case where the first submitted app in aqc queue does not exist,
+      // addressing the issue related to YARN-11708.
+      if (queue == null) {
+        queue = getOrCreateQueueFromPlacementContext(app.getApplicationId(), app.getUser(),
+              app.getQueue(), app.getApplicationPlacementContext(), false);
+      }
+
+      if (queue == null) {
+        String message;
+        if (isAmbiguous(queueName)) {
+          message = "Application " + app.getApplicationId()
```

Comment:
I think here we have some code duplication. We can add the 
`"Application " + app.getApplicationId() + " submitted by user " + app.getUser()`
part to the line 3385